### DISCUSSION TITLE
WiX4 Application after installation crashes immediately

### DISCUSSION THREAD

**Discussion by EmployeeHard** [wixtoolset/issues#7868] [Questions]
I am following the Deployment Dojo Episode 11 HeatWave! The New Hotness for WiX v4.

I can build and run the installer no problem, but the issue I am having is when I go to launch the installed app it closes immediately.
I have a basic console application like in the episode a hello world app that waits for user input after displaying the text so it doesn't close just like in the episode.

I have search on different forums but none of the answers were helpful, the most common culprits for similar issues to mine were missing permissions, I checked and have the permission to install and run the program, or dependencies which I have none for this type of basic application

I followed the steps:
Creating a new  WiX4 MSI Package.
Modify the files according to the guide.
Add the dependency of my project to the WiX project.
Check project buiding configurations made sure it was on x86 just like in the video with all the same options selected

Package.wxs
`<Wix xmlns="http://wixtoolset.org/schemas/v4/wxs">
  <Package Name="WiX_MSI_Installer" Manufacturer="Integer" Version="1.0.0.0" UpgradeCode="ea63b412-0af8-408c-9caf-2c6959c6a4a3">
    <MajorUpgrade DowngradeErrorMessage="!(loc.DowngradeError)" />

    <Feature Id="Main">
      <ComponentGroupRef Id="ExampleComponents" />
    </Feature>
      
  </Package>
</Wix>
`

ExampleComponents.wxs

`<Wix xmlns="http://wixtoolset.org/schemas/v4/wxs">
  <Fragment>
    <ComponentGroup Id="ExampleComponents" Directory="INSTALLFOLDER">
      <Component>
        <File Source="ConsoleApp.exe" />
      </Component>
    </ComponentGroup>
  </Fragment>
</Wix>
`

Folders.wxs
`<Wix xmlns="http://wixtoolset.org/schemas/v4/wxs">
  <Fragment>
    <StandardDirectory Id="ProgramFiles6432Folder">
      <Directory Id="INSTALLFOLDER" Name="!(bind.Property.Manufacturer) !(bind.Property.ProductName)" />
    </StandardDirectory>
  </Fragment>
</Wix>
`

**Comment by chrpai** [comment]
Any error messages in eventviewer?  Any errors displayed in stdout/stderr when you run it from a console window?  You are probably missing a dependency.

**Reply by EmployeeHard** [reply]
Hi, thanks for the reply!

I will go through each of the points you brought up to give more information.

Any error messages in eventviewer?
The event viewer shows the following error.
Description: A .NET application failed.
Application: ConsoleApp.exe
Path: C:\Wix\ConsoleApp.exe
Message: The application to execute does not exist: 'C:\Wix\ConsoleApp.dll'.

Note: That is the correct path to where the installed program should go.

Any errors displayed in stdout/stderr when you run it from a console window?
No errors are displayed, and when I catch the terminal before closing it doesn't display anything.
And when executing the compiled program without the installer it works as it should.

As to the part of missing a dependency if that is the case how can I find out which one it is?
I am working with just a console HelloWorld program that after printing waits for user input before closing, just like in the Deployment Dojo Episode 11 HeatWave! The New Hotness for WiX v4. In that episode no dependencies are brought up apart from adding the ConsoleApp as a dependency to the WiX project which I have done.

Thank you in advance

**Comment by EmployeeHard** [comment]
Hi, thanks for the reply!

I will go through each of the points you brought up to give more information.

Any error messages in eventviewer?
The event viewer shows the following error.
Description: A .NET application failed.
Application: ConsoleApp.exe
Path: C:\Wix\ConsoleApp.exe
Message: The application to execute does not exist: 'C:\Wix\ConsoleApp.dll'.

Note: That is the correct path to where the installed program should go.

Any errors displayed in stdout/stderr when you run it from a console window?
No errors are displayed, and when I catch the terminal before closing it doesn't display anything.
And when executing the compiled program without the installer it works as it should.

As to the part of missing a dependency if that is the case how can I find out which one it is?
I am working with just a console HelloWorld program that after printing waits for user input before closing, just like in the Deployment Dojo Episode 11 HeatWave! The New Hotness for WiX v4. In that episode no dependencies are brought up apart from adding the ConsoleApp as a dependency to the WiX project which I have done.

Thank you in advance

**Comment by chrpai** [comment]
Sounds like you authored the exe but not the dll.  Theres also a few other files .net apps need.
